src/com/owncloud/android/utils/DisplayUtils.java

```diff
@@ -245,6 +258,67 @@ public static Point getScreenSize(Activity caller) {
     }
 
     /**
+     * Determines if user set folder to grid or list view. If folder is not set itself,
+     * it finds a parent that is set (at least root is set).
+     * @param file
+     * @param storageManager
+     * @return
+     */
+    public static boolean isGridView(OCFile file, FileDataStorageManager storageManager){

```

Comment:
I wouldn't do this in a static method. An ```Activity``` or a ```Fragment``` would be a better place.

Even further, did you try to implement the code added to ```FileDisplayActivity``` in ```OCFileListFragment``` instead? That would result in a better structure IMHO. The feature is contained in the scope of ```OCFileListFragment``` responsibility. 

The addition of the menu item in the ```ActionBar``` should be possible also from ```OCFileListFragment```.

Would you give it a try?